### PR TITLE
feat: add gemini-1.5-pro-preview-0409 to vertex ai

### PR DESCRIPTION
**Pull request by cheahjs**
Has gone into public preview: https://console.cloud.google.com/vertex-ai/publishers/google/model-garden/gemini-1.5-pro-preview-0409

### CODE REVIEW

File: litellm/model_prices_and_context_window_backup.json [hunk 1]
```diff
@@ -970,6 +970,17 @@
         "supports_function_calling": true,
         "source": "https://cloud.google.com/vertex-ai/generative-ai/docs/learn/models#foundation_models"
     },
+    "gemini-1.5-pro-preview-0409": {
+        "max_tokens": 8192,
+        "max_input_tokens": 1000000,
+        "max_output_tokens": 8192,
+        "input_cost_per_token": 0,
```
Comment (ishaan-jaff) [inline litellm/model_prices_and_context_window_backup.json:977]: why is cost 0 ?

File: litellm/model_prices_and_context_window_backup.json [hunk 2]
```diff
@@ -970,6 +970,17 @@
...
+        "output_cost_per_token": 0,
+        "litellm_provider": "vertex_ai-language-models",
+        "mode": "chat",
+        "supports_function_calling": false,
```
Comment (ishaan-jaff) [inline litellm/model_prices_and_context_window_backup.json]: `supports_function_calling` should be `True` right ?

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]
left review

File: litellm/model_prices_and_context_window_backup.json (hunk 1 above)
Comment (cheahjs) [reply litellm/model_prices_and_context_window_backup.json:977]: While it's in preview, billing is disabled. Google has announced pricing for the Gemini API version of Gemini 1.5 Pro (https://ai.google.dev/pricing), but not for the Vertex AI version.

File: litellm/model_prices_and_context_window_backup.json (hunk 2 above)
Comment (cheahjs) [reply litellm/model_prices_and_context_window_backup.json]: Updated it, just tried it and it worked, confused it with the Gemini API where the preview didn't have function calling.

File: litellm/model_prices_and_context_window_backup.json (hunk 1 above)
Comment (cheahjs) [reply litellm/model_prices_and_context_window_backup.json:977]: Cloud Next's keynote is ongoing, so they might announce pricing there? 🤷

Comment (cheahjs) [reply litellm/model_prices_and_context_window_backup.json:977]: They've updated the docs, will add costs https://cloud.google.com/vertex-ai/generative-ai/pricing

Asked in Discord as well, but how do you want to map Vertex AI's per-character costs into per-token costs?

Gemini 1.5 Pro
Image Input	$0.00265 / image
Video Input	$0.00265 / second
Text Input	$0.0025 / 1k characters
Audio Input	$0.00025 / second
Text Output	$0.0075 / 1k characters

Comment (ghost) [reply litellm/model_prices_and_context_window_backup.json:977]: we use the openai conversion of 1 token ~= 4 chars

**Comment by ghost** [comment]
Merging PR - as this blocks people from being able to use `gemini-1.5-pro-preview-0409`. will update pricing

**Comment by ishaan-jaff** [comment]
hi @cheahjs can we hop on a call sometime this week. I'd love to learn how we can improve litellm for you. What's the best email to send an invite to ?

If it's easier here's a link to my cal https://calendly.com/d/4mp-gd3-k5k/litellm-1-1-onboarding-chat

**Pull request merged**